scripts/create-release-archive.js

```diff
@@ -0,0 +1,193 @@
+
+  console.log(`Creating release archive for ${newVersion} ${version}`)
+
+  const name = `govuk-prototype-kit-${version}`
+  const releaseArchive = `${name}.zip`
+
+  const repoDir = path.resolve(__dirname, '..')
+  const workdir = fs.mkdtempSync(
+    path.join(os.tmpdir(), `govuk-prototype-kit--release-${version}`)
+  )
+
+  console.log(workdir)
+
+  copyReleaseFiles(repoDir, workdir, { prefix: name, ref: ref })
+
+  // Make the changes we want to make
+  // Currently just removing dev stuff from package.json
+  console.log('Updating package.json')
+  updatePackageJson(path.join(workdir, name, 'package.json'), cleanPackageJson)
+
+  // Create the release archive in the project root (or destDir)
+  zipReleaseFiles({
+    cwd: workdir, file: path.join(argv.dest || repoDir, releaseArchive), prefix: name
+  })
+
+  console.log(`Saved release archive to ${argv.dest ? argv.dest : ''}${releaseArchive}`)
+
+  // Clean up
+  fs.rmdirSync(workdir, { recursive: true })
+}
+
+function updatePackageJson (file, updater) {
+  let pkg
+  pkg = JSON.parse(fs.readFileSync(file, { encoding: 'utf8' }))
+  pkg = updater(pkg)
+  fs.writeFileSync(file, JSON.stringify(pkg, null, 2) + os.EOL, { encoding: 'utf8' })
+  // update package-lock.json to match
+  child_process.execSync('npm install', { cwd: path.dirname(file), encoding: 'utf8', stdio: 'inherit' })
+}
+
+function cleanPackageJson (pkg) {
+  // remove dev dependencies
+  delete pkg.devDependencies
+
+  // remove config for dev dependencies
+  delete pkg.jest
+  delete pkg.standard
+
+  // remove dev scripts
+  pkg.scripts = Object.fromEntries(
+    Object.entries(pkg.scripts)
+      .filter(([name]) => packageJsonScriptsInclude.includes(name))
+  )
+
+  return pkg
+}
+
+function getReleaseVersion (ref) {
+  if (!ref) {
+    const packageVersion = require('../package').version
+    if (isNewVersion(packageVersion)) {
+      return packageVersion
+    }
+  }
+
+  ref = ref || 'HEAD'
+
+  const versionString = child_process.execSync(`git describe --tags ${ref}`, { encoding: 'utf8' }).trim()
+  const version = versionString.slice(1) // drop the initial 'v'
+
+  return version
+}
+
+function isNewVersion (version) {
+  return !!child_process.spawnSync(
+    'git', ['rev-parse', `v${version}`]
+  ).status
+}
+
+function copyReleaseFiles (src, dest, { prefix, ref }) {
+  // We are currently using the export-ignore directives in .gitattributes to
+  // decide which files to include in the release archive, so the easiest way
+  // to copy all the release files is `git archive`
+  child_process.execSync(
+    `git archive --format=tar --prefix="${prefix}/" ${ref} | tar -C ${dest} -xf -`,
+    { cwd: src }
+  )
+}
+
+function zipReleaseFiles ({ cwd, file, prefix }) {
+  zipCreate(
+    {
+      cwd: cwd,
+      file: file,
+      exclude: path.join(prefix, 'node_modules', '*')
+    },
+    prefix
+  )
+}
+
+function zipCreate ({ cwd, file, exclude }, files) {
```

Comment:
would this be better named as `createZip`?

Reply:
I think it's fine either way... I named it this way round because I was imaging it could one day be replaced by a Node library called `zip` with a `create` method...

Reply:
I agree that it would be good to use a zip library.  It's an internal function so I don't think we'll struggle to change this name or implementation after the merge/release.